iotfunctions/anomaly.py

```diff
@@ -167,32 +117,30 @@ def execute(self, df):
             dfe_orig = dfe_orig.reset_index(level=[0]).sort_index()
 
             # minimal time delta for merging
-            mindelta = dfe_orig.index.to_series().diff().min()
-            if mindelta == dt.timedelta(seconds = 0) or pd.isnull(mindelta):
-                mindelta = pd.Timedelta('5 seconds')
-
-            # compute meandelta for upsampling
-            meandelta = dfe_orig.index.to_series().diff().mean()
-            if meandelta == dt.timedelta(seconds = 0) or pd.isnull(meandelta):
-                meandelta = mindelta
+            mindelta = minDelta(dfe_orig)
 
-            logger.info('Timedelta:' + str(mindelta) + ',' + str(meandelta))
+            logger.info('Timedelta:' + str(mindelta))
 
-            # upsample original per entity dataframe and compute the gap frame
-            upsampled_na = dfe_orig.resample(meandelta).apply(custom_resampler)
-            dfe = upsampled_na.where(upsampled_na.isna(), 0).fillna(1)
+            # count the timedelta in seconds between two events
+            timeSeq = dfe.index.values - dfe.index[0].to_datetime64()
+            temperature = np.gradient(timeSeq)  # we look at the gradient for anomaly detection
 
             # interpolate gaps - data imputation
-            Size = dfe[[self.input_item]].to_numpy().size
+            dfe[[self.input_item]] = temperature
+            Size = temperature.size
+            dfe = dfe.interpolate(method='time')
 
             # one dimensional time series - named temperature for catchyness
-            temperature = dfe[[self.input_item]].to_numpy().reshape(-1,)
+            #temperature = dfe[[self.input_item]].fillna(0).to_numpy().reshape(-1,)
 
-            logger.debug('NoDataAnomaly: ' + str(entity) + ', ' + str(self.input_item) + ', ' + str(self.windowsize) + ', ' +
+            logger.info('Spectral: ' + str(entity) + ', ' + str(self.input_item) + ', ' + str(self.windowsize) + ', ' +
                          str(self.output_item) + ', ' + str(self.windowoverlap) + ', ' + str(temperature.size))
 
-            if temperature.size > self.windowsize:
-                logger.debug(str(temperature.size) + str(self.windowsize))
+            if temperature.size <= self.windowsize:
+                logger.info(str(temperature.size) + ' <= ' + str(self.windowsize))
+                df_copy.loc[[entity]] = 0.0001
+            else:
+                logger.info(str(temperature.size) + str(self.windowsize))
                 # Fourier transform:
                 #   frequency, time, spectral density
                 freqsTS, timesTS, SxTS = signal.spectrogram(temperature, fs = self.frame_rate, window = 'hanning',
```

Comment:
name variable accordingly(  frequency, time, spectral density)

Reply:
it's now 
frequency_temperature, time_series_temperature, spectral_density_temperature